src/software/thunderscope/chicker/chicker.py

```diff
@@ -0,0 +1,227 @@
+        grid.addWidget(self.createRadio("No Auto"), 1, 0)
+        grid.addWidget(self.createRadio("Auto Kick"), 1, 1)
+        grid.addWidget(self.createRadio("Auto Chip"), 1, 2)
+        grid.addWidget(self.createSlider("Geneva Slider", 1, 5, 1), 2, 0, 1, 3)
+        grid.addWidget(self.createSlider("Power Slider", 1, 100, 10), 3, 0, 1, 3)
+        self.setLayout(grid)
+        self.grid = grid
+
+        # state of radio buttons in order NoAuto, AutoKick, AutoChip
+        self.radioCheckable = [True, True, True]
+        self.charged = False
+        self.geneva_value = 1
+        self.power_value = 1
+
+    # all the buttons, radios, and sliders are in a group box --> vbox (vertical alignment)
+    def createButton(self, text):
+        groupBox = QGroupBox()
+        button = QPushButton(text)
+        groupBox.setStyleSheet("color: black")
+        button.setCheckable(True)
+        if text != "Charge":
+            button.setCheckable(False)
+            button.setStyleSheet("background-color: Grey")
+        vbox = QVBoxLayout()
+        vbox.addWidget(button)
+        vbox.addStretch(1)
+        groupBox.setLayout(vbox)
+        return groupBox
+
+    def createRadio(self, text):
+        groupBox = QGroupBox()
+        radio = QRadioButton(text)
+        groupBox.setStyleSheet("color: black")
+        # this is so that the button is properly visible (it looked very dim otherwise)
+        radio.setStyleSheet("background-color: white")
+        self.radio_buttons.addButton(radio)
+        vbox = QVBoxLayout()
+        vbox.addWidget(radio)
+        vbox.addStretch(1)
+        groupBox.setLayout(vbox)
+        return groupBox
+
+    def createSlider(self, text, minVal, maxVal, tickSpacing):
+        groupBox = QGroupBox(text)
+        slider = QSlider(Qt.Horizontal)
+        slider.setMinimum(minVal)
+        slider.setMaximum(maxVal)
+        slider.setTickPosition(QSlider.TicksBothSides)
+        slider.setTickInterval(tickSpacing)
+        groupBox.setStyleSheet("color: white")
+        vbox = QVBoxLayout()
+        vbox.addWidget(slider)
+        vbox.addStretch(1)
+        groupBox.setLayout(vbox)
+        return groupBox
+
+    def refresh(self):
+        # set 'Charge'/'Discharge' text based on self.charged value
+        for button in self.grid.parentWidget().findChildren(QPushButton):
+            if self.charged:
+                if button.text() == "Charge":
+                    button.setText("Discharge")
+            elif not self.charged:
+                if button.text() == "Discharge":
+                    button.setText("Charge")
+
+        # iterate over all boxes - possibly the only way to access the widgets
+        # grid --> groupBox --> button/radio/slider
+        for item in self.grid.parentWidget().findChildren(QGroupBox):
+
+            # update slider values
+            for slider in item.findChildren(QSlider):
+                # if slider.valueChanged():
```

Comment:
I think you can avoid iterating here if you stored the button before assigning it to the widget

So instead of 
~~~
grid.addWidget(self.createButton(..), 0, 0)
~~~
 we do
~~~
self.charge_button = self.createButton("Charge")
grid.addWidget(self.charge_button, 0, 0)
~~~

Then you can just access the sliders/buttons and radios here without iterating and matching text